### DISCUSSION TITLE
1.9.1如何配置数据库连接地址

### DISCUSSION THREAD

**Discussion by GDFboy** [hhyo/Archery#1920] [General]
# 该项目本身的mysql数据库地址
DATABASES = {
    "default": {
        **env.db(),
        **{
            "DEFAULT_CHARSET": "utf8mb4",
            "CONN_MAX_AGE": 50,
            "OPTIONS": {
                "init_command": "SET sql_mode='STRICT_TRANS_TABLES'",
                "charset": "utf8mb4",
            },
            "TEST": {
                "NAME": "test_archery",
                "CHARSET": "utf8mb4",
            },
        },
    }
}

这是要修改哪呢

**Comment by nick2wang** [comment]
这里配置
https://github.com/hhyo/Archery/blob/029ebab887e18642f056fb562fd309a9776788c0/archery/settings.py#L17

**Reply by GDFboy** [reply]
密码在哪配置呢

**Reply by nick2wang** [reply]
> 密码在哪配置呢

mysql://user:password@127.0.0.1:3306/archery

**Reply by GDFboy** [reply]
我已经按着你说的给配置上了，但是在表结构初始化的时候报错（/opt/venv4archery/lib/python3.9/site-packages/django/core/management/commands/makemigrations.py:143: RuntimeWarning: Got an error checking a consistent migration history performed for database connection 'default': (2005, "Unknown MySQL server host 'mysql' (2)")
  warnings.warn(
No changes detected in app 'sql'
）

配置
    DATABASE_URL=(str, "mysql://root:123\#AAaa@127.0.0.1:3306/archery"),
    CACHE_URL=(str, "redis://127.0.0.1:6379/0?PASSWORD=123456#aaa"),

**Reply by GDFboy** [reply]
是还有别的地方需要修改吗

**Reply by nick2wang** [reply]
这里是url的写法，密码包含特殊符号需要进行编码处理：
```
import urllib

...
    DATABASE_URL=(str, f"mysql://root:{urllib.parse.quote('123#AAaa')}@127.0.0.1:3306/archery"),
    CACHE_URL=(str, f"redis://:{urllib.parse.quote('123456#aaa')}@127.0.0.1:6379/0"),
...
```

**Reply by GDFboy** [reply]
提示NameError: name 'urllib' is not defined

**Reply by nick2wang** [reply]
> 提示NameError: name 'urllib' is not defined

要先引入
`import urllib`

**Comment by hhyo** [comment]
docker部署修改这个文件
https://github.com/hhyo/Archery/blob/master/src/docker-compose/.env

具体怎么修改，env文件内都有引导链接
https://django-environ.readthedocs.io/en/latest/quickstart.html#usage
